internal/engine/filter/convert.go

```
@@ -174,9 +173,8 @@ func generateNodeAsString(node *parser.Node) (string, error) {
 
 func generateLeafAsString(leaf *parser.Leaf) (string, error) {
 	needEscape := leaf.Function != parser.FunctionQuery && !leaf.IsRaw
-	query := joinStrings(needEscape, leaf.Args...)
+	query, err := joinStrings(needEscape, leaf.Args...)
```

You need to return the error here:

```
query, err := joinStrings(...)
if err != nil {
    return "", err
}
```